### DISCUSSION TITLE
Allow specification of config within package.json

### DISCUSSION THREAD

**Discussion by ghost** [rome/tools#4543] [Suggestions]
I find most of the default setting of Rome to be wonderful, with that being the case pretty much the only thing I need to enable and configure is the `vcs` settings. In turn I end up with a rather tiny `rome.json` almost seems wasteful having a whole file just for it.

Hence my proposition, allow me to place the 4 line configuration object in my `package.json`.

**Comment by ematipico** [comment]
It's a good suggestion, but Rome is not there yet - it's not able to read the `package.json` yet.

Although, this proposal comes with two big downsides: 
- you would lose the possibility to use the JSON schema
- the `package.json` would risk becoming very big (this is a personal preference)

**Reply by ghost** [reply]
In response to your downsides:

- You can always write the json in another file with the schema, then paste into. 
- Only if you have a lot of configuration to do.

 I do as much of my config in the `package.json` as possible because it's a file I rarely need to open after setup and it cleans up my file tree a little. 

As you said it's a preference, having the option would be good.

**Comment by Conaclos** [comment]
Could you share your current _Rome_ configuration? This could help to identify better defaults?

**Comment by ghost** [comment]
Here it is: 

``` rome.json
{
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "linter": {
    "rules": {
      "recommended": true,
      "style": {
        "noCommaOperator": "off"
      }
    }
  }
}
```

It's mostly default except for enabling VCS to allow the "useIgnoreFile" which saves me specifying ignoring things I've already specified once. This is an absolutely wonderful feature and I thank you for adding it.

Also to the linter configuration I've added `"noCommaOperator": "off"` as I don't like extracting to click handler functions if it's just two functions and doesn't expand to two lines. Unless someone can give me a good reason why I should always extract things like the example below to a click handler function?

``` tsx
<button
  type="button"
  onClick={() => (setSelected(i), setOpen(!open))}
  onKeyDown={() => (setSelected(i), setOpen(!open))}
  className={`${styles.button} h-auto input`}
>
  {option.name}
</button>
```

**Reply by Conaclos** [reply]
In the future, _vcs_ could be enabled by default? @ematipico

Otherwise, I think disabling the comma operator is a good default.

To come back to the subject of this thread, I am not a big fan of embedded configuration in _package.json_. I do not know if it is still a common idiom in the community.

**Reply by ematipico** [reply]
> In the future, _vcs_ could be enabled by default? @ematipico
> 

What do you mean precisely? To enable the ignore file by default, and choose git?

> Otherwise, I think disabling the comma operator is a good default.
> 
> To come back to the subject of this thread, I am not a big fan of embedded configuration in _package.json_. I do not know if it is still a common idiom in the community.

The majority of the tools support package.json too: prettier, jest, lint-staged, etc. The initial version of Rome supported that too. I'm not a big fan too, but I don't see any reason why not add the support.

**Reply by Conaclos** [reply]
> What do you mean precisely? To enable the ignore file by default, and choose git?

We could detect which _vcs_ is used (`.gitignore` => git, .hgignore => hg, ...). In the case of the detection of several _vcs_ (e.g. `.gitignore` and `.hgignore`) they could be all enabled. This could reauire to change the config to support multi-vcs projects. Or we could expect explicit configuration when several _vcs_ are detected.

**Reply by ghost** [reply]
> > What do you mean precisely? To enable the ignore file by default, and choose git?
> 
> We could detect which _vcs_ is used (`.gitignore` => git, .hgignore => hg, ...). In the case of the detection of several _vcs_ (e.g. `.gitignore` and `.hgignore`) they could be all enabled. This could reauire to change the config to support multi-vcs projects. Or we could expect explicit configuration when several _vcs_ are detected.

This would be a good solution, saves the developer specifying something which can be detected quite quickly. 

> Otherwise, I think disabling the comma operator is a good default.

I agree the comma operator is more of a personal preference and makes sense to leave off by default.

> The majority of the tools support package.json too: prettier, jest, lint-staged, etc. The initial version of Rome supported that too. I'm not a big fan too, but I don't see any reason why not add the support.

Best to encompass as many developer preferences as possible without it being too much work.
